src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2381,7 +2381,7 @@ LogicalResult ONNXCastOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
   ShapedType inputType = input().getType().dyn_cast<ShapedType>();
   if (!inputType) {
-    return emitError("Non-shaped input type");
+    return success();
```

Comment:
Nit: in most places we test for `.getType().isa<RankedTensorType>()` but here its for `getType().dyn_cast<ShapedType>()`. 

It seems to be the same thing, but I actually don't know the difference between ShapedType vs RankedTensorType. 

The same test for ShapedType occurs elsewhere too